Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -209,12 +209,12 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
             if ([error isKindOfClass:[NSNull class]])
                 return;
 
-            callOnMainThread([parent = _parent, layer = WTFMove(layer), error = WTFMove(error)] {
+            callOnMainThread([parent = RefPtr { _parent.get() }, layer = WTFMove(layer), error = WTFMove(error)] {
```

Comment:
I think this will break if KVO notifications are ever sent off the main thread, as the WeakPtr will assert. But maybe KVO is main thread-only?

Reply:
Good point, KVO observers can be called on whatever thread changes the value.